### PR TITLE
Gnutls v3.5.0 benchmark

### PR DESCRIPTION
**Pull request by nettrino**

### CODE REVIEW

File: gnutls-v3.5.0/Readme.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+Finds a SEGV error during parsing of a corrupted certificate, in
```
Comment (kcc) [inline gnutls-v3.5.0/Readme.md:1]: Same here: please indicate how much time is needed to find this bug. 
If it's several seconds, it's probably not that interesting.

File: gnutls-v3.5.0/Readme.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+[GnuTLS](https://www.gnutls.org/) v.3.5.0. Fixed in [da4c7a39](https://gitlab.com/gnutls/gnutls/commit/da4c7a393d273076af4f650f6cb1fd6329078824).
+
+POC
+===
+```
+../BUILD/bin/certtool --verify-chain --load-ca-certificate ca_chain.pem --infile full_pem_chain.pem
+```
+
+(To attempt replication with the respective libFuzzer driver, run with
+`ASAN_OPTIONS=detect_leaks=0` and a starting corpus of certificates in PEM
```
Comment (kcc) [inline gnutls-v3.5.0/Readme.md:11]: Why  ASAN_OPTIONS=detect_leaks=0? 
If there is also a leak that it easy to find, this makes it a bad benchmark :(

Comment (kcc) [reply gnutls-v3.5.0/Readme.md:11]: Do you provide the "starting corpus" here?

File: gnutls-v3.5.0/driver.cc [hunk 3]
```diff
@@ -0,0 +1,228 @@
+#include <stdlib.h>
```
Comment (kcc) [inline gnutls-v3.5.0/driver.cc:1]: same here: why do you need the driver?

File: gnutls-v3.5.0/gnutls_3.5.0_no_sig.patch [hunk 4]
```diff
@@ -0,0 +1,244 @@
+diff -ruN gnutls-vanilla/lib/pubkey.c gnutls_3.5.0_sign/lib/pubkey.c
+--- gnutls-vanilla/lib/pubkey.c	2016-07-06 14:02:33.298994233 -0400
++++ gnutls_3.5.0_sign/lib/pubkey.c	2016-07-06 15:49:54.201136838 -0400
+@@ -1,7 +1,7 @@
+ /*
+  * GnuTLS public key support
+  * Copyright (C) 2010-2012 Free Software Foundation, Inc.
+- *
```
Comment (kcc) [inline gnutls-v3.5.0/gnutls_3.5.0_no_sig.patch:8]: This patch has too many comment-only diffs. Why? Can this be made smaller?

File: gnutls-v3.5.0/Readme.md (hunk 2 above)
Comment (nettrino) [reply gnutls-v3.5.0/Readme.md:11]: I believe these leaks are due in the internals of GnuTLS and not related to the bug or the driver. I will double check this and also update the PR to address the comments!

File: gnutls-v3.5.0/gnutls_3.5.0_no_sig.patch (hunk 4 above)
Comment (nettrino) [reply gnutls-v3.5.0/gnutls_3.5.0_no_sig.patch:8]: Probably my VIM config trimming whitespace :). Will address this

File: gnutls-v3.5.0/Readme.md (hunk 2 above)
Comment (kcc) [reply gnutls-v3.5.0/Readme.md:11]: If there is a leak that happens frequently we can't use it as a benchmark, as it will exhaust all memory quickly.

Comment (nettrino) [reply gnutls-v3.5.0/Readme.md:11]: Agreed.

**Pull request closed without merging**